src/v3/test/integration/byol.test.tsx

```diff
@@ -0,0 +1,125 @@
+import loginFooBundle from '@okta/mocks/labels/json/login_foo.json';
+import countryFooBundle from '@okta/mocks/labels/json/country_foo.json';
+import { waitFor, cleanup } from '@testing-library/preact';
+import { rest } from 'msw';
+import { setupServer, SetupServerApi } from 'msw/node';
+import { setup } from './util';
+
+describe('byol', () => {
+  beforeEach(() => {
+    // language is cached after first test so much disabling caching for this test
+    jest.spyOn(global, 'localStorage', 'get').mockReturnValue({
+      length: 0,
+      clear: () => jest.fn(),
+      getItem: () => null,
+      setItem: () => jest.fn(),
+      key: () => null,
+      removeItem: () => jest.fn(),
+    });
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('loading custom language', () => {
+    const server: SetupServerApi = setupServer();
+    beforeAll(() => {
+      server.listen();
+    });
+
+    beforeEach(() => {
+      server.use(
+        rest.get('http://localhost:1234/mocks/labels/json/login_foo.json', async (_, res, ctx) => res(
+          ctx.status(200),
+          ctx.json(loginFooBundle),
+        )),
+        rest.get('http://localhost:1234/mocks/labels/json/country_foo.json', async (_, res, ctx) => res(
+          ctx.status(200),
+          ctx.json(countryFooBundle),
+        )),
+      );
+    });
+
+    afterEach(() => {
+      server.resetHandlers();
+    });
+
+    afterAll(() => {
+      server.close();
+    });
+
+    it('should load custom language with assets.baseUrl and assets.languages set and with navigator.languages containing language', async () => {
+      const navigatorLanguagesSpy = jest.spyOn(global, 'navigator', 'get');
+      navigatorLanguagesSpy.mockReturnValue(
+        { languages: ['foo'] } as unknown as Navigator,
+      );
+      const { container, findByText } = await setup({
+        mockResponse,
+        widgetOptions: {
+          assets: {
+            baseUrl: 'http://localhost:1234/mocks',
+            languages: ['foo'],
+          },
+        },
+      });
+      await waitFor(async () => {
+        await findByText(/Set up foo authentication/);
+        await findByText(/Foonited States/);
+      });
+      expect(container).toMatchSnapshot();
+    });
+
+    it('should load custom language with "language" option defined, and assets.baseUrl is set to a path containing the language assets', async () => {
+      const { container, findByText } = await setup({
+        mockResponse,
+        widgetOptions: {
+          language: 'foo',
+          assets: { baseUrl: 'http://localhost:1234/mocks' },
+        },
+      });
+      await waitFor(async () => {
+        await findByText(/Set up foo authentication/);
+        await findByText(/Foonited States/);
+      });
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe('loading default language', () => {
+    it('should not load custom language with "language" option set', async () => {
```

Comment:
can we add a bit more clarification here? maybe call out that specifically the reason it's not loading custom language is because baseUrl is not set